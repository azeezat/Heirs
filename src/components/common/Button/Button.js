import React from 'react';
import PropTypes from 'prop-types';
import {NavLink} from 'react-router-dom';
import './Button.scss'

const renderColour=(colour)=>{
    switch (colour.toString().toLowerCase()) {
        case 'green':
            return 'green-btn'
        case 'transparent':
            return 'transparent-btn'
        case 'dark blue':
            return 'dark-blue-btn'
        default:
            break;
    }
}

const Button = ({text,colour, to}) => {
    return (
        <NavLink to={`${to?to:'/'}`}>
            <input type="button" className={`buttons ${renderColour(colour)}` } value={text}/>
        </NavLink>
    );
};

Button.propTypes = {
    colour: PropTypes.oneOf(['green', 'transparent','dark blue']),
    text: PropTypes.string.isRequired
};

export default Button;