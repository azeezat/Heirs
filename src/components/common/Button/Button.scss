@import '../../../_utilities/variables';

.buttons{
    cursor: pointer;
    color: $white;
    font-size: 1.3rem;
    padding: 9px 26px;
    border-radius: 50px;
    letter-spacing: 1px;
}
.green-btn{
    border: 2px solid $green; 
    background: $green;
}

.transparent-btn{
    border: 2px solid $darkBlue; 
    background: none;
    color: $blue;
}

.dark-blue-btn{
    border: 2px solid $darkBlue; 
    background-color: $darkBlue;
    color: $white;
}