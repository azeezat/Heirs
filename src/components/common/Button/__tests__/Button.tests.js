import React from 'react'
import { shallow } from 'enzyme'
import Button from '../Button';
/**
 * @description Button
 */

describe('Button component', () => {
    let props={
        colour:'green',
        text: 'Login'
    }
    let shallowWrapper = shallow(
        <Button {...props}/>
    )

    it('renders `Button` properly', () => {
        expect(shallowWrapper).toMatchSnapshot();
    })
})