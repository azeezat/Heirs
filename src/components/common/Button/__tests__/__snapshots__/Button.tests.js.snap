// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders \`Button\` properly 1`] = `ShallowWrapper {}`;
